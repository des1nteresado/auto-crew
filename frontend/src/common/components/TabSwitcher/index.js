import React from 'react';
import PropTypes from 'prop-types';

import { Switch, Route } from 'react-router-dom';

const TabSwitcher = ({ tabs, baseUrl, param }) => {
  return (
    <Switch>
      {tabs.map((route) => {
        const { path, Component } = route;

        return (
          <Route
            path={`${baseUrl}${path}${param ? `/:${param}` : ''}`}
            render={(props) => <Component {...props} />}
            key={path}
          />
        );
      })}
    </Switch>
  );
};
TabSwitcher.defaultProps = {
  baseUrl: '/',
  param: '',
};

TabSwitcher.propTypes = {
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string.isRequired,
    })
  ).isRequired,
  baseUrl: PropTypes.string,
  param: PropTypes.string,
};

export default React.memo(TabSwitcher);
