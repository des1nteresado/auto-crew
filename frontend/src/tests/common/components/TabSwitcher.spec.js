import React from 'react';
import { shallow } from 'enzyme';

import TabSwitcher from 'common/components/TabSwitcher';

const defaultProps = {
  tabs: [
    {
      path: 'path1',
      Component: <div>component1</div>,
    },
    {
      path: 'path2',
      Component: <div>component2</div>,
    },
  ],
};

const setUp = (props) => shallow(<TabSwitcher {...defaultProps} {...props} />).dive();

describe('TabSwitcher', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it('renders TabSwitcher component correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
