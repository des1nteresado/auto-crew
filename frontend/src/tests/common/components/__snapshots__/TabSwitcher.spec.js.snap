// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabSwitcher renders TabSwitcher component correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
